src/button/button.js

```diff
@@ -5,61 +5,110 @@
 
 'use strict';
 
+import Model from '../model.js';
 import Controller from '../controller.js';
+import ControllerCollection from '../controllercollection.js';
+import Icon from '../icon/icon.js';
+import IconView from '../icon/iconview.js';
 
 /**
- * The basic button controller class.
+ * The Button controller class. It uses {@link ui.icon.Icon} component
+ * to display an icon.
+ *
+ *		const model = new Model( {
+ *			label: 'Bold',
+ *			isEnabled: true,
+ *			isOn: false,
+ *			icon: 'bold',
+ *			iconAlign: 'LEFT'
+ *		} );
+ *
+ *		// An instance of Button with a label and an icon.
+ *		new Button( model, new ButtonView() );
+ *
+ * See {@link ui.button.ButtonView}.
  *
  * @memberOf ui.button
  * @extends ui.Controller
  */
 export default class Button extends Controller {
+	/**
+	 * Creates an instance of {@link ui.button.Button} class.
+	 *
+	 * @param {ui.button.ButtonModel} model Model of this Button.
+	 * @param {ui.View} view View of this Button.
+	 */
 	constructor( model, view ) {
 		super( model, view );
 
+		view.model.bind( 'label', 'isOn', 'isEnabled' ).to( model );
+
+		if ( model.icon ) {
+			view.model.bind( 'icon', 'iconAlign' ).to( model );
+		}
+
 		view.model.on( 'click', () => model.fire( 'execute' ) );
 	}
+
+	/**
+	 * @inheritDoc
+	 */
+	init() {
+		if ( this.model.icon ) {
+			this.collections.add( new ControllerCollection( 'children' ) );
+
+			const iconModel = new Model();
+			iconModel.bind( 'name', 'align' ).to( this.model, 'icon', 'iconAlign' );
+
+			this.add( 'children', new Icon( iconModel, new IconView() ) );
+		}
+
+		return super.init();
+	}
 }
 
 /**
- * The basic button model interface.
+ * The Button component {@link ui.Model} interface.
```

Comment:
Are we sure that we need to write Button uppercase? This may become confusing later on.